packages/widget_toolkit_pin/lib/src/lib_pin_code_with_biometrics/blocs/pin_code_bloc.dart

```diff
@@ -125,69 +119,30 @@ class PinCodeBloc extends $PinCodeBloc {
   }
 
   @override
-  ConnectableStream<List<BiometricsAuthType>>
-      _mapToAvailableBiometricsState() => _$checkAvailableBiometricsEvent
-          .switchMap((value) => biometricAuthenticationService
-              .availableBiometrics
-              .asResultStream())
-          .setResultStateHandler(this)
-          .whereSuccess()
-          .publishReplay();
-
-  @override
-  ConnectableStream<bool> _mapToIsAuthenticatedWithBiometricsState() =>
-      _$requestBiometricAuthEvent
-          .switchMap(
-              (value) => _biometricAuthentication(value).asResultStream())
-          .setResultStateHandler(this)
-          .whereSuccess()
-          .publishReplay(maxSize: 1);
-
-  Future<bool> _biometricAuthentication(String localizedReason) async {
-    var success = await biometricAuthenticationService.authenticate(
-        localizedReason.isEmpty ? enterPinWithBiometrics : localizedReason);
-
-    return success;
-  }
-
-  @override
-  ConnectableStream<bool> _mapToIsPinCodeVerifiedState() => _$autoSubmitEvent
-      .switchMap((pinCode) => encryptAndVerify(pinCode).asResultStream())
-      .setResultStateHandler(this)
-      .whereSuccess()
-      .publishReplay(maxSize: 1);
-
-  Future<bool> encryptAndVerify(String pinCode) async {
-    var encryptedPin = await pinCodeService.encryptPinCode(pinCode);
-    var verifiedPin = await pinCodeService.verifyPinCode(encryptedPin);
-    return verifiedPin;
-  }
-
-  @override
-  Stream<BiometricsMessage?> _mapToBiometricsDialogState() =>
-      _$setBiometricsEvent
-          .switchMap((event) => biometricAuthenticationService
-              .enableBiometrics(event.enabled, event.localizedReason)
-              .asResultStream())
-          .setResultStateHandler(this)
-          .whereSuccess()
-          .doOnData((event) {
-        if (event == BiometricsMessage.enabled) {
-          checkBiometricsEnabled();
+  ConnectableStream<bool> _mapToShowBiometricsButtonState() => _digitsCountState
+      .asyncMap((digitsCount) async {
+        final int storedPinLength = await pinCodeService.getPinLength();
```

Comment:
Here also the variables `storedPinLength, isCorrectPin`  could be written only with final without the type